src/lib/input/input-container.ts

```diff
@@ -257,11 +266,16 @@ export class MdInputContainer implements AfterContentInit {
   }
   private _hintLabel = '';
 
-  /** Text or the floating placeholder. */
+  /** Whether the placeholder should always float, never float or float as the user types. */
   @Input()
-  get floatingPlaceholder(): boolean { return this._floatingPlaceholder; }
-  set floatingPlaceholder(value) { this._floatingPlaceholder = coerceBooleanProperty(value); }
-  private _floatingPlaceholder: boolean = true;
+  get floatingPlaceholder() { return this._floatingPlaceholder; }
+  set floatingPlaceholder(value: MD_INPUT_PLACEHOLDER_TYPES) {
+    if (value && MD_INPUT_PLACEHOLDER_VALUES.indexOf(value) === -1) {
```

Comment:
👍 - Notice that invalid values for the `floatPlaceholder` attribute will automatically fallback to `auto` due to the getters [here](https://github.com/DevVersion/material2/blob/ae203c3ae49175b3630b9f69aba608b57cd7efb4/src/lib/input/input-container.ts#L254-L257)